Automattic/jetpack
# Promote with Blaze: Purge cache on Atomic sites when site visibility changes

This PR adds an improvement for Atomic sites that clears the eligibility cache every time the user changes the site's visibility status. 

This is needed because currently the cache is set to 24h, meaning that the user will not be aware of the feature until the cache is expired and the eligibility call is triggered again.

Also, on Atomic sites, we no longer had the "Promote with Blaze" option, as we do for Simple Sites. This PR changes this to be more in line with the behavior we have on Simple and re-adds the item only for Atomic sites.





Related to https://github.com/Automattic/wp-calypso/issues/95621

## Proposed changes:

* Purge the transient on Atomic sites when the site visibility changes
* Bring back the quick link in posts list on Atomic sites

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Use the Jetpack beta and update wpcomsh and Jetpack
* Go to an Atomic site and make it private (WP-Admin > Settings > General)
* Notice that there's no menu item for WP-Admin > Tools > Advertising
* Notice that in WP-Admin > Posts there's no "Promote with Blaze" quick link
* Set the site to coming soon and repeat the steps
* Set the site to public - you should now get the Tools >Advertising menu and the Posts > Promote with Blaze quick link
* Set the visibility back and forth to make sure that the cache is indeed cleared properly.
